### PR TITLE
[opengarage] Adds invert option to switch

### PR DESCRIPTION
**Pull request by digitaldan**
Signed-off-by: digitaldan <dan@digitaldan.com>

This adds the ability to invert the switch state so that OFF=OPEN and ON=Closed.  This allows the state to be more aligned with other close/locking things like Z-wave locks, while maintaining backwards compatibility for existing users.

### CODE REVIEW

File: bundles/org.openhab.binding.opengarage/src/main/java/org/openhab/binding/opengarage/internal/OpenGarageChannelConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+package org.openhab.binding.opengarage.internal;
+
+/**
+ * The OpenGarageChannelConfiguration class contains fields mapping thing configuration parameters.
+ *
+ * @author Dan Cunningham - Initial contribution
+ */
+
+public class OpenGarageChannelConfiguration {
+    boolean invert = false;
```
Comment (cpmeister) [inline bundles/org.openhab.binding.opengarage/src/main/java/org/openhab/binding/opengarage/internal/OpenGarageChannelConfiguration.java:22]:
```suggestion
    public boolean invert = false;
```

File: bundles/org.openhab.binding.opengarage/src/main/java/org/openhab/binding/opengarage/internal/OpenGarageChannelConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+ */
+package org.openhab.binding.opengarage.internal;
+
+/**
+ * The OpenGarageChannelConfiguration class contains fields mapping thing configuration parameters.
+ *
+ * @author Dan Cunningham - Initial contribution
+ */
+
+public class OpenGarageChannelConfiguration {
```
Comment (cpmeister) [inline bundles/org.openhab.binding.opengarage/src/main/java/org/openhab/binding/opengarage/internal/OpenGarageChannelConfiguration.java:21]: Please add `@NonNullByDefault`

File: bundles/org.openhab.binding.opengarage/src/main/java/org/openhab/binding/opengarage/internal/OpenGarageChannelConfiguration.java (hunk 1 above)
Comment (digitaldan) [reply bundles/org.openhab.binding.opengarage/src/main/java/org/openhab/binding/opengarage/internal/OpenGarageChannelConfiguration.java]: Good catch

File: bundles/org.openhab.binding.opengarage/src/main/java/org/openhab/binding/opengarage/internal/OpenGarageChannelConfiguration.java (hunk 2 above)
Comment (digitaldan) [reply bundles/org.openhab.binding.opengarage/src/main/java/org/openhab/binding/opengarage/internal/OpenGarageChannelConfiguration.java]: 👍

File: bundles/org.openhab.binding.opengarage/README.md [hunk 3]
```diff
@@ -23,13 +23,13 @@ As a minimum, the IP address is needed:
...
-| status-rollershutter | Rollershutter | Door status (DOWN = Closed, UP = Open)                 |
-| vehicle              | String        | Report vehicle presence from the OpenGarage controller |
+| channel              | type          | description                                                                           |
+|----------------------|---------------|---------------------------------------------------------------------------------------|
+| distance             | Number:Length | Distance reading from the OpenGarage controller (default in cm)                       |
+| status-switch        | Switch        | Door status (OFF = Closed, ON = Open), set "invert=true" on channel to invert switch  |
+| status-contact       | Contact       | Door status (Open or Closed)                                                          |
+| status-rollershutter | Rollershutter | Door status (DOWN = Closed, UP = Open)                                                |
+| vehicle-status       | Number        | Report vehicle presence (0=Not Detected, 1=Detected, 2=Unknown)                       |
```
Comment (cpmeister) [inline bundles/org.openhab.binding.opengarage/README.md:32]: Why is the channel name changed? Was this a bug in the readme?

Comment (digitaldan) [reply bundles/org.openhab.binding.opengarage/README.md:32]: I changed that in another PR that was merged , but I must of missed committing the README change in it ( i probably did a `commit -s ./src` )  .   The string channel is depreciated, although still there for backwards compatibility, in favor of this number type which has state options in the channel xml definition.

**Comment by digitaldan** [comment]
Thank @cpmeister , i very much appreciate your attention to details and valuable feedback.

**Pull request merged**